### PR TITLE
util: refine Row.GetDatum when tp is MyDecimal

### PR DESCRIPTION
**Pull request by XuHuaiyu**
Refer #6077 
This commit moves the dereference of *MyDecimal from 
Row.GetDecimal to Row.GetDatum to avoid the copy of MyDecimal
when calling Row.GetMyDecimal.

PTAL @coocood @zz-jason

### CODE REVIEW

File: util/chunk/chunk.go [hunk 1]
```diff
@@ -620,7 +619,8 @@ func (r Row) GetDatum(colIdx int, tp *types.FieldType) types.Datum {
 		}
 	case mysql.TypeNewDecimal:
 		if !r.IsNull(colIdx) {
-			d.SetMysqlDecimal(r.GetMyDecimal(colIdx))
+			dec := *r.GetMyDecimal(colIdx)
```
Comment (shenli) [inline util/chunk/chunk.go]: Is there any other place calls Row.GetMyDecimal()?

**Comment by coocood** [comment]
LGTM

File: util/chunk/chunk.go (hunk 1 above)
Comment (XuHuaiyu) [reply util/chunk/chunk.go]: Yes, but only max/min will hold the value directly now.

**Pull request merged**